File: adventofcode/2021/day2/day21.cpp
#include <bits/stdc++.h>
using namespace std;
/*
part 1
*/
int main() {
    freopen("inputs/input1.txt", "r", stdin);
    freopen("outputs/output1.txt", "w", stdout);
    string input;
    long long depth = 0, hor = 0;
    while (getline(cin, input)) {
        int pos = input.find(" ");
        string direction = input.substr(0, pos);
        int magnitude = stoi(input.substr(pos + 1));
        if (direction == "down") {
            depth += magnitude;
        } else if (direction == "up") {
            depth -= magnitude;
        } else if (direction == "forward") {
            hor += magnitude;
        }
    }
    cout<<depth*hor<<endl;
}

File: adventofcode/2021/day02/day21.py
from collections import namedtuple
if __name__ == '__main__':
    with open("inputs/input1.txt", "r") as f:
        arr = list(map(lambda x: (x.split()[0], int(x.split()[1])), f.read().splitlines()))
        hor = sum(magnitude for direction, magnitude in arr if direction in ['forward'])
        depth = sum(magnitude*(1 if direction=='down' else -1) for direction, magnitude in arr if direction in ['up', 'down'])
        print(hor*depth)
